sync-todo/v2/client/cpp/controllers/home_controller.cpp

```diff
@@ -0,0 +1,193 @@
+#include "home_controller.hpp"
+
+/** Conveniently provide a vertical wrapper for a text label and a component, centered. */
+ftxui::Component VWrap(const std::string& name, const ftxui::Component& component) {
+  return Renderer(component, [name, component] {
+    return ftxui::vbox({
+      hbox(ftxui::text(name)) | ftxui::hcenter | size(ftxui::WIDTH, ftxui::EQUAL, 18),
+      ftxui::separator(),
+      component->Render() | ftxui::hcenter,
+      });
+  });
+}
+
+HomeController::HomeController(AppState *appState): Controller(ftxui::Container::Vertical({})), _appState(appState), _homeControllerState(new HomeControllerState) {
```

Comment:
There is a memory leak here. The allocated HomeControllerState is never `deleted` in the destructor.
But you should basically never use `new`.
Use smart pointers: make_unique() is responsible for allocating and the unique_ptr will delete when it goes out of scope.
However, in this case, there's probably no reason to dynamically allocate the HomeControllerState and you can just declare it as a member of the class.